packages/@icon-magic/config-reader/src/index.ts

```diff
@@ -0,0 +1,68 @@
+import * as path from 'path';
+import { Config } from './config';
+import { IconSetHashInterface } from '@icon-magic/icon-models/src/interface';
+import {
+  exists,
+  isDirectory
+} from '@icon-magic/icon-models/dist/src/utils/files';
+
+const glob = require('glob');
+
+const CONFIG_FILES = [
+  '.iconrc.json',
+  '.iconrc.js',
+  '.iconrc',
+  'iconrc.json',
+  'iconrc.js',
+  'iconrc'
+];
+
+export function getIconSetwithConfigs(
+  inputPaths: string[]
+): IconSetHashInterface {
+  const validInputDirs = processPaths(inputPaths);
+  if (!validInputDirs.length) {
+    throw new Error('Input paths are not found!');
+  }
+  let configFiles: string[] = [];
+  for (let inputDir of validInputDirs) {
+    configFiles = configFiles.concat(findFilesSync(inputDir, CONFIG_FILES));
+  }
+  // dedup the set of config files
+  configFiles = [...new Set(configFiles)];
```

Comment:
Rather than go through this extra ceremony, try:

```
let configFiles: Set<string> = new Set();
for (let inputDir of validInputDirs) {
  configFiles.add(findFilesSync(inputDir, CONFIG_FILES));
}
const iconSet = new Config([...configFiles]);
```